### PR TITLE
Feature/276

### LINKED ISSUE #276: Integration tests change values in the database for localhost or live server
## Description
When running the integration tests more than once, the number of failed tests increases from 46 to 100+. The integration tests access the actual database used in production or what is used locally and change it. Since the tests fail, the database is changed, so it doesn't work as intended anymore. 

**Possible Suggested Solution**  
Change, so the tests run in an temporary database that will be deleted after the tests have been completed.

**This issue will be expanded upon when further information is discovered.**

### PR DESCRIPTION
**Pull request by ltranu20**
# Description

- Added proper dispose methods to fixtures adding new records to the database.
- Fixed an integration test incorrectly changing the password of an user to their ResetToken.
- Added more properties to some fixtures (reduce the number of magic strings in code).
- Changed the relation between AspNetUsers and Setting to a one-to-one relationship.
- Added a hack to delete corresponding Settings record when deleting a user (the relation in the database is currently flawed but fixing this is out of scope for this PR).

Fixes #276

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

it just works stop asking
(tested on iOS device)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have Acceptance Tested this on an iOS device
- [ ] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: GirafRest.IntegrationTest/Setup/AccountFixture.cs [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
 namespace GirafRest.IntegrationTest.Setup
 {
-    public class AccountFixture
+    public class AccountFixture : IDisposable
     {
         public string GuardianUsername;
         public string Citizen2Username;
         public string Citizen1Username;
         public string Password;
+        public string BrandNewPassword;
```
Comment (NikolajKrogh) [inline GirafRest.IntegrationTest/Setup/AccountFixture.cs:12]: Hvorfor hedder det BrandNew- og ikke bare NewPassword?

File: GirafRest.IntegrationTest/Setup/WeekTemplateFixture.cs [hunk 2]
```diff
@@ -1,41 +0,0 @@
-﻿using Newtonsoft.Json.Linq;
```
Comment (NikolajKrogh) [inline GirafRest.IntegrationTest/Setup/WeekTemplateFixture.cs:1]: Hvorfor har i slettet WeekTemplateFixture?

**Review by NikolajKrogh — COMMENTED** [review]
Jeg har kun sat et par kommentarer. Super fedt i har omdøbt nogle funktioner og variabler, det gjorde det dejlig nemt at læse:)

File: GirafRest.IntegrationTest/Setup/UserFixture.cs [hunk 3]
```diff
@@ -56,6 +56,20 @@ public UserFixture()
                                       {{ 'hexColor': '#FFFFFF', 'day': 7}}]}}";
             RawImage = "ÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿñÇÿÿõ×ÿÿñÇÿÿÿÿÿÿÿ" +
```
Comment (danielalexlarsen) [inline GirafRest.IntegrationTest/Setup/UserFixture.cs:57]: har det altid stået sådan der? 😮

**Comment by danielalexlarsen** [comment]
> Jeg har kun sat et par kommentarer. Super fedt i har omdømt nogle funktion og variabler, det gjorde det dejlig nemt at læse:)

+1 på den her, rigtig fedt :D har testet på emulatoren, LGTM

**Review by danielalexlarsen — APPROVED** [review]

File: GirafRest.IntegrationTest/Setup/AccountFixture.cs (hunk 1 above)
Comment (ltranu20) [reply GirafRest.IntegrationTest/Setup/AccountFixture.cs:12]: Navnet blev bare kopieret fra den gamle 🪄magiske streng🪄 i de eksisternede tests.

File: GirafRest.IntegrationTest/Setup/WeekTemplateFixture.cs (hunk 2 above)
Comment (ltranu20) [reply GirafRest.IntegrationTest/Setup/WeekTemplateFixture.cs:1]: WeekTemplate tests blev slettet for nogle commits siden, så den stod bare og samlede støv (samt også en mindre fixture jeg skal tilføje dipose methods til 😌)

File: GirafRest.IntegrationTest/Setup/UserFixture.cs (hunk 3 above)
Comment (ltranu20) [reply GirafRest.IntegrationTest/Setup/UserFixture.cs:57]: Yup, ingen ide om hvorfor xd

**Review by Jaffenheimer — APPROVED** [review]
Changes has been made to accomodate @NikolajKrogh's review approving this

**Pull request merged**